vendor/github.com/ethereum/go-ethereum/whisper/notificationserver/server.go

```diff
@@ -0,0 +1,72 @@
+package notificationserver
+
+import (
+	"crypto/ecdsa"
+
+	"github.com/ethereum/go-ethereum/p2p"
+	whisper "github.com/ethereum/go-ethereum/whisper/whisperv5"
+	"github.com/status-im/status-go/geth/params"
+)
+
+// TO DO(rgeraldes)
+// Restore sessions after reviewing the storage model
+
+const (
+	datadirDefaultSessionStore = "sessionstore"
+	topicServerDiscovery       = "/server/discover"
+	topicServerAcceptance      = "/server/accept"
+)
+
+// NotificationServer -
+type NotificationServer struct {
+	*whisper.Server                   // whisper server
+	Store                             // session store
+	Notifier                          // firebase client
+	serverID        string            // server ID
+	protocolKey     *ecdsa.PrivateKey // shared protocol secret
+}
+
+// Init initializes the notification server
+func (s *NotificationServer) Init(w *whisper.Whisper, config *params.WhisperConfig) error {
```

Comment:
Should we consider letting `NotificationServer` hold a slice of `[]Notifier` to allow future addition of notification systems later, eg one for slack?